lib/cequel/metal/keyspace.rb

```diff
@@ -226,6 +227,32 @@ def execute_with_options(statement, options={})
       end
 
       #
+      # Wraps the prepare statement in the default retry strategy
+      #
+      # @param statement [String,Statement] statement to prepare
+      # @return [Cassandra::Statement::Prepared] the prepared statement
+      #
+      def prepare_statement(statement)
+        retries = max_retries
+        cql = case statement
+              when Statement
+                statement.cql
+              else
+                statement
+              end
+        begin
+          client.prepare(cql)
+        rescue Cassandra::Errors::NoHostsAvailable,
+               Cassandra::Errors::ExecutionError
+          clear_active_connections!
+          raise if retries == 0
+          retries -= 1
+          sleep(retry_delay)
+          retry
+        end
+      end
```

Comment:
Could we extract this and the execute retry logic into a retry method that takes a block?